packages/amplify-category-function/src/provider-utils/awscloudformation/utils/lambda-layer-cloudformation-template.ts

```diff
@@ -0,0 +1,130 @@
+import { Fn, DeletionPolicy, Refs } from 'cloudform';
+import _ from 'lodash';
+import Lambda from 'cloudform-types/types/lambda';
+import { Permission, LayerParameters, getLayerMetadataFactory, LayerMetadata } from './layerParams';
+
+function generateLayerCfnObjBase() {
+  const cfnObj = {
```

Comment:
Use `cloudForm` classes not pure javascript objects, this way we don't get any validation at compiler level, true for all the cfn resource constructions in the PR